### DISCUSSION TITLE
Is it possible to query the depth buffer in a post processing shader?

### DISCUSSION THREAD

**Discussion by david4shure** [bevyengine/bevy#10324] [Q&A]
I am currently working on this shader and trying to port it to bevy and wgsl 
https://github.com/BarthPaleologue/volumetric-atmospheric-scattering/blob/main/src/glsl/atmosphericScattering.glsl

I notice in it's invocation that it is invoked by the babylonjs project as a post processing effect
(see here https://github.com/BarthPaleologue/volumetric-atmospheric-scattering/blob/main/src/ts/starter.ts#L35 and here https://github.com/BarthPaleologue/volumetric-atmospheric-scattering/blob/main/src/ts/atmosphericScattering.ts#L30)

In order for this shader to function I will need to sample the depth texture and the screen texture at a specific coordinate
(see here https://github.com/BarthPaleologue/volumetric-atmospheric-scattering/blob/main/src/glsl/atmosphericScattering.glsl#L178-L180). 

My question is this: Is it possible to sample the depth buffer in a post processing effect in bevy? I tried to do this and couldn't get it working. And if it is NOT possible, How should I proceed? As far as I can tell I am unable to sample the screen texture in the prepass shader.

**Comment by JMS55** [comment]
Add a DepthPrepass component to your camera, and then bind the depth texture from ViewPrepassTextures iirc is the component name. You might also be able to bind the main depth texture directly, without using the prepass, idr though.

**Reply by david4shure** [reply]
Ok, so I have added the DepthPrepass component to my camera. Now, just to make sure I understand, I need to query for the ViewPrepassTextures component, and bind the depth texture field from the ViewPrepassTextures component to my post processing shader? 

And this would mean I would use the bind group rather than these prepass utils to fetch the depth texture (see here https://github.com/bevyengine/bevy/blob/main/assets/shaders/show_prepass.wgsl#L27)?

**Reply by JMS55** [reply]
Yes
